src/event/metric.rs

```diff
@@ -32,13 +33,13 @@ pub struct MetricSeries {
     pub tags: Option<MetricTags>,
 }
 
-pub type MetricTags = BTreeMap<String, String>;
+pub type MetricTags = BTreeMap<ImStr, String>;
```

Comment:
The value part could be `ImStr` too, it's never modified in-place.

Reply:
Yessir, I agree. The compiler error list here is brutal so I'm changing one thing at a time.